src/app/digital/internal/views/components/BUFGateAssembler.ts

```diff
@@ -0,0 +1,110 @@
+    public readonly size = V(1, 1);
+    public readonly img: SVGDrawing;
+
+    protected readonly sim: DigitalSim;
+
+    protected portAssembler: PortAssembler;
+
+    public constructor(circuit: CircuitInternal, view: CircuitView, selections: SelectionsManager, sim: DigitalSim) {
+        super(circuit, view, selections);
+
+        this.sim = sim;
+
+        this.img = view.options.getImage("buf.svg")!;
+
+        this.portAssembler = new PortAssembler(circuit, view, selections, {
+            "outputs": () => ({ origin: V(0.5, 0), dir: V(1, 0) }),
+            "inputs": () => ({ origin: V(-0.5, 0), dir: V(-1, 0) }),
+        });
+    }
+
+    private assembleLine(gate: Schema.Component) {
+        const { defaultBorderWidth, selectedBorderColor, defaultBorderColor } = this.options;
+
+        const { inputPortGroups } = this.circuit.getObjectInfo("BUFGate") as DigitalComponentInfo;
+
+        // Get current number of inputs
+        const numInputs = [...this.circuit.getPortsForComponent(gate.id).unwrap()]
+            .map((id) => this.circuit.getPortByID(id).unwrap())
+            .filter((p) => ((p) && (inputPortGroups.includes(p.group)))).length;
+
+        const dy = (numInputs-1)/2*(0.5 - defaultBorderWidth/2);
+        const y1 = -dy - defaultBorderWidth/2;
+        const y2 =  dy + defaultBorderWidth/2;
+
+        const x = -(this.size.x - defaultBorderWidth)/2;
+
+        const transform = this.view.componentTransforms.get(gate.id)!;
+
+        const selected = this.selections.has(gate.id);
+
+        return new Line(
+            transform.toWorldSpace(V(x, y1)),
+            transform.toWorldSpace(V(x, y2)),
+            new Style(undefined,
+                (selected ? selectedBorderColor : defaultBorderColor),
+                defaultBorderWidth,
+            ),
+        );
+    }
+
+    private assembleImage(gate: Schema.Component) {
+        const selected = this.selections.has(gate.id);
+        const tint = (selected ? this.options.selectedFillColor : undefined);
+        return new SVGPrim(this.img, this.size, this.view.componentTransforms.get(gate.id)!, tint);
+    }
+
+    public assemble(gate: Schema.Component, ev: unknown) {
+        const transformChanged = /* use ev to see if our transform changed */ true;
+        const selectionChanged = /* use ev to see if we were de/selected */ true;
+        const portAmtChanged   = /* use ev to see if the number of ports changed */ true;
+
+        if (!transformChanged && !selectionChanged && !portAmtChanged)
+            return;
+
+        if (transformChanged) {
+            // Update transform
+            this.view.componentTransforms.set(gate.id, new Transform(
+                V(gate.props.x ?? 0, gate.props.y ?? 0),
+                this.size,
+                (gate.props.angle ?? 0),
+            ));
+        }
+
+        this.portAssembler.assemble(gate, ev);
+
+        const [prevLine, prevImg] = (this.view.componentPrims.get(gate.id) ?? []);
```

Comment:
BUFGate doesn't have a line since it doesn't change the number of inputs, so please simplify accordingly